### PR TITLE
In table of contents, include line above every h2 item in list

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

**Comment by montehurd** [comment]
👍

**Comment by joewalsh** [comment]
on develop branch, i get:
```
2016-06-20 14:40:01.223 Wikipedia Debug[49847:17130947] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x7ff663305a10 Wikipedia.WMFTableOfContentsHeader:0x7ff663304c40.trailing == UILabel:0x7ff6633052d0.trailing + 20>",
    "<NSLayoutConstraint:0x7ff663305b00 UILabel:0x7ff6633052d0.leading == Wikipedia.WMFTableOfContentsHeader:0x7ff663304c40.leading + 20>",
    "<NSLayoutConstraint:0x7ff65e4ede40 Wikipedia.WMFTableOfContentsHeader:0x7ff663304c40.width == 8>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x7ff663305a10 Wikipedia.WMFTableOfContentsHeader:0x7ff663304c40.trailing == UILabel:0x7ff6633052d0.trailing + 20>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

**Comment by joewalsh** [comment]
and I get the same error on this branch:
```
Debug[51377:17156095] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x7fb5b1467070 Wikipedia.WMFTableOfContentsHeader:0x7fb5b17f6d80.trailing == UILabel:0x7fb5b7710a70.trailing + 20>",
    "<NSLayoutConstraint:0x7fb5b147baa0 UILabel:0x7fb5b7710a70.leading == Wikipedia.WMFTableOfContentsHeader:0x7fb5b17f6d80.leading + 20>",
    "<NSLayoutConstraint:0x7fb5b1460150 Wikipedia.WMFTableOfContentsHeader:0x7fb5b17f6d80.width == 8>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x7fb5b1467070 Wikipedia.WMFTableOfContentsHeader:0x7fb5b17f6d80.trailing == UILabel:0x7fb5b7710a70.trailing + 20>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

**Comment by montehurd** [comment]
@joewalsh gotcha thanks for checking

File: Wikipedia/Code/WMFTableOfContentsItem.swift [hunk 1]
```diff
@@ -40,6 +40,10 @@ public enum TableOfContentsItemType {
 public enum TableOfContentsBorderType {
     case Default
     case FullWidth
+    case DefaultTopOnly
+    case FullWidthTopOnly
+    case DefaultBottomOnly
+    case FullWidthBottomOnly
     case None
 }
```
Comment (montehurd) [inline Wikipedia/Code/WMFTableOfContentsItem.swift]: @joewalsh 
Looks like we only use None, Default and DefaultTopOnly. Thoughts on maybe removing the unused cases?

**Pull request merged**